gusto/linear_solvers.py

```diff
@@ -69,12 +63,35 @@ class CompressibleSolver(TimesteppingSolver):
     :arg quadrature degree: tuple (q_h, q_v) where q_h is the required
     quadrature degree in the horizontal direction and q_v is that in
     the vertical direction
-    :arg params (optional): solver parameters
+    :arg solver_parameters (optional): solver parameters
+    :arg overwrite_solver_parameters: boolean, if True use only the
+    solver_parameters that have been passed in, if False then update
+    the default solver parameters with the solver_parameters passed in.
     """
 
-    def __init__(self, state, quadrature_degree=None, params=None):
-
-        self.state = state
+    solver_parameters = {
+        'pc_type': 'fieldsplit',
+        'pc_fieldsplit_type': 'schur',
+        'ksp_type': 'gmres',
+        'ksp_max_it': 100,
+        'ksp_gmres_restart': 50,
+        'pc_fieldsplit_schur_fact_type': 'FULL',
+        'pc_fieldsplit_schur_precondition': 'selfp',
+        'fieldsplit_0': {'ksp_type': 'preonly',
+                         'pc_type': 'bjacobi',
+                         'sub_pc_type': 'ilu'},
+        'fieldsplit_1': {'ksp_type': 'preonly',
+                         'pc_type': 'gamg',
+                         'mg_levels': {'ksp_type': 'chebyshev',
+                                       'ksp_chebyshev_esteig': True,
+                                       'ksp_chebyshev_esteig_random': True,
```

Comment:
PETSc changing their parameter names and not advertising it well?  Well I never...